plugins/permission-backend/src/service/router.ts

```diff
@@ -116,41 +175,41 @@ const handleRequest = async (
   });
 
   return Promise.all(
-    requests.map(({ id, resourceRef, ...request }) =>
-      policy.handle(request, user).then(decision => {
-        if (decision.result !== AuthorizeResult.CONDITIONAL) {
-          return {
-            id,
-            ...decision,
-          };
-        }
-
-        if (!isResourcePermission(request.permission)) {
-          throw new Error(
-            `Conditional decision returned from permission policy for non-resource permission ${request.permission.name}`,
-          );
-        }
+    requests.map(async ({ id, resourceRef, ...request }) => {
+      const cacheKey = cacheEnabled
+        ? `${user?.identity.userEntityRef}-${Buffer.from(
+            JSON.stringify(request),
+          ).toString('base64')}`
+        : '';
 
-        if (decision.resourceType !== request.permission.resourceType) {
-          throw new Error(
-            `Invalid resource conditions returned from permission policy for permission ${request.permission.name}`,
-          );
+      if (cacheEnabled && cache) {
+        const cachedResponse = await cache.get<string>(cacheKey);
+        if (cachedResponse) {
+          return JSON.parse(cachedResponse);
```

Comment:
Just a note here that when persisting "blind" data like this, it's important to remember to update the cache keys when making changes - otherwise when you change the response shape in any way you're at risk of reading cached data that's on the old format and crashing stuff all over for a while.

There's no validation already of that format, that we can leverage to just try/catch a parse of it and only return if it is valid?